bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareBridgeHandler.java

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.surepetcare.internal.handler;
+
+import static org.openhab.binding.surepetcare.internal.SurePetcareConstants.*;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.surepetcare.internal.AuthenticationException;
+import org.openhab.binding.surepetcare.internal.SurePetcareAPIHelper;
+import org.openhab.binding.surepetcare.internal.discovery.SurePetcareDiscoveryService;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareBridgeConfiguration;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareDevice;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareHousehold;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcarePet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SurePetcareBridgeHandler} is responsible for handling the bridge things created to use the Sure Petcare
+ * API. This way, the user credentials may be entered only once.
+ *
+ * It also spawns 2 background polling threads to update the more static data (topology) and the pet locations at
+ * different time intervals.
+ *
+ * @author Rene Scherer - Initial Contribution
+ */
+@NonNullByDefault
+public class SurePetcareBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SurePetcareBridgeHandler.class);
+
+    private final SurePetcareAPIHelper petcareAPI;
+    private @Nullable ScheduledFuture<?> topologyPollingJob = null;
```

Comment:
This explicit ` = null` is not needed without it the same would happen

Reply:
done